Util/Helper.php

```diff
@@ -4,53 +4,77 @@
 
 use Doctrine\Common\Annotations\AnnotationReader;
 use Padam87\CronBundle\Annotation\Job;
+use ReflectionAttribute;
+use ReflectionClass;
+use ReflectionException;
 use Symfony\Component\Console\Application;
 use Symfony\Component\Console\Command\LazyCommand;
 use Symfony\Component\Console\Input\InputInterface;
 
 class Helper
 {
-    private $application;
-    private $annotationReader;
+    private Application $application;
+    private AnnotationReader $annotationReader;
 
     public function __construct(Application $application, AnnotationReader $annotationReader)
     {
         $this->application = $application;
         $this->annotationReader = $annotationReader;
     }
 
+    /**
+     * @throws ReflectionException
+     */
     public function createTab(InputInterface $input, ?array $config = null): Tab
     {
+        $attributesSupported = class_exists('ReflectionAttribute');
+
+        $group = $input->hasOption('group') ? $input->getOption('group') : null;
+
         $tab = new Tab();
 
-        foreach($this->application->all() as $command) {
+        foreach ($this->application->all() as $command) {
             $commandInstance = $command instanceof LazyCommand
                 ? $command->getCommand()
                 : $command;
 
-            $annotations = $this->annotationReader->getClassAnnotations(new \ReflectionClass($commandInstance));
+            $reflectionClass = new ReflectionClass($commandInstance);
 
-            foreach ($annotations as $annotation) {
-                if ($annotation instanceof Job) {
-                    $group = $input->hasOption('group') ? $input->getOption('group') : null;
+            $attributesOrAnnotations = array_merge(
```

Comment:
This section needs rework. Never a good idea to mix objects of different class in an array.

I'd check for annotations first, and in case it is not found check attributes.

In case of attribute use the aforementioned `Job::fromAttribute` static method, to convert it to a `Job`.

Reply:
On second thought, lets flip that... attributes first is probably better, and shouldn't have side effects.

Reply:
Also, $attribute->newInstance() should work.

Reply:
> Also, $attribute->newInstance() should work.

This ain't gonn work either, because newInstance will pass the arguments in the way call_user_func works, and no keys are preserved this way, to when only defining `dayOfWeek`, the value of it will be assigned to the `minute` property.